ext/spl/config.m4

```diff
@@ -1,4 +1,5 @@
-PHP_NEW_EXTENSION(spl, php_spl.c spl_functions.c spl_engine.c spl_iterators.c spl_array.c spl_directory.c spl_exceptions.c spl_observer.c spl_dllist.c spl_heap.c spl_fixedarray.c, no,, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1)
-PHP_INSTALL_HEADERS([ext/spl], [php_spl.h spl_array.h spl_directory.h spl_engine.h spl_exceptions.h spl_functions.h spl_iterators.h spl_observer.h spl_dllist.h spl_heap.h spl_fixedarray.h])
+PHP_NEW_EXTENSION(spl, php_spl.c spl_functions.c spl_engine.c spl_iterators.c spl_array.c spl_directory.c spl_exceptions.c spl_observer.c spl_dllist.c spl_heap.c spl_fixedarray.c spl_cachediterable.c, no,, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1)
+PHP_INSTALL_HEADERS([ext/spl], [php_spl.h spl_array.h spl_directory.h spl_engine.h spl_exceptions.h spl_functions.h spl_iterators.h spl_observer.h spl_dllist.h spl_heap.h spl_fixedarray.h spl_cachediterable.h])
 PHP_ADD_EXTENSION_DEP(spl, pcre, true)
 PHP_ADD_EXTENSION_DEP(spl, standard, true)
+PHP_ADD_EXTENSION_DEP(spl, json, true)
```

Comment:
Is there specific reason for wanting this? I don't think existing structures are json serializable, so is there specific motivation for adding it and incurring this dependency?

Reply:
https://wiki.php.net/rfc/always_enable_json passed in PHP 8.0.

Before that, adding a dependency on json would complicate package maintenance, and the fact that json could be compiled as a shared extension that was potentially disabled would mean that adding it was impractical (can't have an always-on extension depend on an extension that can be disabled)

Some applications may depend on the existing structures json behavior, e.g. I've used `new ArrayObject($array)` in the past to get associative arrays always encoded as `{}`, even empty ones